<?xml version="1.0"?>
<launch>
<!-- 
    <rosparam param="propeller_position_x"> [2.373776, 2.373776,  -2.373776,  -2.373776]</rosparam>
    <rosparam param="propeller_position_y"> [1.027135,  -1.027135, 1.027135,  -1.027135]</rosparam>
    <rosparam param="propeller_position_z"> [0.318237, 0.318237, 0.318237, 0.318237]</rosparam> -->
    <!-- <rosparam param="propeller_num"> 6 </rosparam> -->
<!--     <rosparam param="propeller_K_positive"> [0.025, 0.025, 0.025, 0.025]</rosparam>
    <rosparam param="propeller_K_negtive"> [0.01, 0.01, 0.01, 0.01]</rosparam>
    <rosparam param="propeller_max_rotation"> [100, 100, 100, 100]</rosparam>
    <rosparam param="propeller_min_rotation"> [5, 5, 5, 5]</rosparam>
    <rosparam param="propeller_max_alfa"> [3.32,   0.0174, 3.32, 0.0174  ]</rosparam>
    <rosparam param="propeller_min_alfa"> [-0.0174,-3.32,  -0.0174,-3.32]</rosparam>
    <rosparam param="propeller_delta_alfa"> [0.157, 0.157, 0.157, 0.157]</rosparam>
    <rosparam param="propeller_delta_rotation"> [30,30,30,30]</rosparam>
    <rosparam param="propeller_max_thrust"> [250,250,250,250]</rosparam>
    <rosparam param="propeller_min_thrust"> [1,1,1,1]</rosparam>
    <rosparam param="propeller_index"> [1, 1, 1, 1]</rosparam>
    <rosparam param="max_target_force"> [250, 250, 250, 250]</rosparam>
    <rosparam param="min_target_force"> [0, 0, 0, 0]</rosparam>

    <rosparam param="ctl_mode"> 0 </rosparam>   -->
      <!-- mode 0 joystick; mode 1 autodp-->
    <!-- <node name = "goal_pub" pkg="virtual_robotx" type = "goal_pub" output = "screen"/> -->
    <node name= "gps_convert_a" pkg="multi_dp" type = "gps_converter" output = "screen">
        <param name = "topic_sub" value = "/defender_a/sensors/gps/gps/fix"/>
        <param name = "topic_pub_UTMx" value = "/defender_a/sensors/gps/gps/UTMx"/>
        <param name = "topic_pub_UTMy" value = "/defender_a/sensors/gps/gps/UTMy"/>
    </node>

    <node name= "gps_convert_b" pkg="multi_dp" type = "gps_converter" output = "screen">
        <param name = "topic_sub" value = "/defender_b/sensors/gps/gps/fix"/>
        <param name = "topic_pub_UTMx" value = "/defender_b/sensors/gps/gps/UTMx"/>
        <param name = "topic_pub_UTMy" value = "/defender_b/sensors/gps/gps/UTMy"/>
    </node>



    <node name= "eular_convert_a" pkg="multi_dp" type = "eular_angle" output = "screen">
        <param name = "topic_sub" value = "/defender_a/sensors/imu/imu/data"/>
        <param name = "topic_pub_yaw" value = "/defender_a/sensors/yaw"/>
    </node>


    <node name= "eular_convert_b" pkg="multi_dp" type = "eular_angle" output = "screen">
        <param name = "topic_sub" value = "/defender_b/sensors/imu/imu/data"/>
        <param name = "topic_pub_yaw" value = "/defender_b/sensors/yaw"/>
    </node>

<!--     <node name = "eular_convert" pkg = "virtual_robotx" type ="eular_angle"/>
    <node name="pid_coordinate" pkg="virtual_robotx" type="pid_coordinate_v" output="screen"/>
    <node name="UTMx_controller" pkg="pid" type="controller" ns= "pid" output="screen">
        <param name="Kp" value="5.0" />
        <param name="Ki" value="0.2" />
        <param name="Kd" value="2" />
        <param name="upper_limit" value="200" />
        <param name="lower_limit" value="-200" />
        <param name="windup_limit" value="10" />
        <param name="topic_from_controller" value="/controller_fx" />
        <param name="topic_from_plant" value="/float64_zero" />
        <param name="setpoint_topic" value="/vrx/station_keeping/x_r" />
        <param name="max_loop_frequency" value="100.0" />
        <param name="min_loop_frequency" value="100.0" />
    </node>

    <node name="UTMy_controller" pkg="pid" type="controller" ns= "pid" output="screen">
        <param name="Kp" value="3" />
        <param name="Ki" value="0.3" />
        <param name="Kd" value="01" />
        <param name="upper_limit" value="200" />
        <param name="lower_limit" value="-200" />
        <param name="windup_limit" value="10" />
        <param name="topic_from_controller" value="/controller_fy" />
        <param name="topic_from_plant" value="/float64_zero" />
        <param name="setpoint_topic" value="/vrx/station_keeping/y_r" />
        <param name="max_loop_frequency" value="100.0" />
        <param name="min_loop_frequency" value="100.0" />
    </node>

    <node name="head_controller" pkg="pid" type="controller" ns= "pid" output="screen">
        <param name="Kp" value="10.0" />
        <param name="Ki" value="0.5" />
        <param name="Kd" value="10" />
        <param name="upper_limit" value="100" />
        <param name="lower_limit" value="-100" />
        <param name="windup_limit" value="10" />
        <param name="topic_from_controller" value="/controller_mz" />
        <param name="topic_from_plant" value="/wamv/sensors/imu/imu/yaw" />
        <param name="setpoint_topic" value="/vrx/station_keeping/head" />
        <param name="max_loop_frequency" value="100.0" />
        <param name="min_loop_frequency" value="100.0" />
    </node>
    <node name="target_rotation" pkg="virtual_robotx" type="target_rotation" output="screen"/>
    <node name="thrust_pub" pkg="virtual_robotx" type="thrust_pub" output="screen"/>
 -->
   

</launch>